components/script/dom/htmlformelement.rs

```diff
@@ -321,7 +321,37 @@ impl HTMLFormElement {
 
         result.push_str(&*format!("\r\n--{}--", boundary));
 
-        return result;
+        result
+    }
+
+    // https://html.spec.whatwg.org/multipage/#text/plain-encoding-algorithm
+    fn encode_plaintext(&self, form_data: &mut Vec<FormDatum>) -> String {
+        // Step 1
+        let mut result = String::new();
+
+        // Step 2
+        let encoding = self.pick_encoding();
+
+        // Step 3
+        let charset = &*encoding.whatwg_name().unwrap();
+
+        for entry in form_data.iter_mut() {
+            // Step 4
+            if entry.name == "_charset_" && entry.ty == "hidden" {
+                entry.value = FormDatumValue::String(DOMString::from(charset.clone()));
+            }
+
+            // Step 5
+            if entry.ty == "file" {
+                entry.value = FormDatumValue::String(DOMString::from(entry.value_str()));
+            }
+
+            // Step 6
+            result.push_str(&*format!("{}={}\r\n", entry.name, entry.value_str()));
+        }
+
+        // Step 7
```

Comment:
Yes, because the spec may change on us without warning. This documents exactly which part of the algorithms that we're implementing with the following code.